src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -219,6 +220,33 @@
      */
     PortalAgentType getPortalAgentType();
 
+    /**
+     * Gets the portal agent map of this world.
+     * Keys represent dimensions names.
+     * Values represent destination world names.
+     *
+     * @return The portal agent map of this world
+     */
+    Map<String, String> getPortalAgents();
```

Comment:
I'm gonna be blunt here, I know what's been said in the past, but I _do not_ like this being a `Map<String, String>`, when the rest of the API tries to use `World` and `WorldProperties` as much as possible.

Don't get me wrong, I understand the reasoning. The idea is that it mirrors the config file, which might have invalid entries that you don't wish to lose. However, the rest of the API doesn't expose the config as strings, so we shouldn't here.

In my view, this should be a `Map<WorldProperties, PortalType>`, or something to that effect, dynamically gemerated when requested. Invalid entries should just be ignored, this should be a representation of what the running server will act upon, and **not** what is in the config file.

Reply:
@dualspiral Completely fair points but it is dangerous to hand back a map reference with WorldProperties in it as stale references can be destructive.

Maybe a Map<PortalType, String> as a compromise?

Reply:
My original thought is that we'd use a `WeakHashMap`... but then I look at the javadocs and what it really should say is the keys are the portal types, not the dimensions. 

So, sure.